### PR TITLE
Add DimensionfulSpinVectorTag and change RadiusCompute to type Scalar<DataVector>

### PR DESCRIPTION
**Pull request by MarloMo**
## Proposed changes

The DimensionfulSpinVectorTag and DimensionfulSpinVectorTagCompute calculate the dimensionful spin angular momentum vector of a Strahlkorper.

Change the RadiusCompute to a type Scalar DataVector in Tags.hpp. 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/ApparentHorizons/Tags.hpp
Comment (wthrowe) [inline src/ApparentHorizons/Tags.hpp]: You can't (usefully) move a const object.  Either remove the `std::move` or change `radius` to a non-const non-reference.  Generally the latter is preferred, but it doesn't matter for compute tags and I think we usually take their arguments by const reference even if it would be bad style normally.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/ApparentHorizons/Tags.hpp
Comment (wthrowe) [inline src/ApparentHorizons/Tags.hpp]: This should return by reference.

**Review by kidder — COMMENTED** [review]
I have no further requested changes

File: src/ApparentHorizons/Tags.hpp
Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp]: I'll remove `std::move`.

**Comment by wthrowe** [comment]
Sorry, I should request more thorough tests for these things.  Please write a test using this tag in  a DataBox.  Something roughly like
```c++
void test_dimensionful_spin_vector_compute_tag() noexcept {
  const auto box = db::create<db::AddSimpleTags<DependencyTag1, etc...>,
                              db::AddComputeTags<DimensionfulSpinVectorCompute<Frame>>(
      dependency_1_value, etc...);
  CHECK(db::get<DimensionfulSpinVector>(box) == ...);
}
```
and call that function from inside the existing SPECTRE_TEST_CASE.  If there are easy values that give a known result for the spin vector you can use those, or you can calculate the right-hand side of the CHECK comparison by calling the `StrahlkorperGr::spin_vector` free function.

File: src/ApparentHorizons/Tags.hpp
Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp]: @wthrowe, should I make a test for all the compute tags that I have merged into Tags.hpp or should I only make this test for the DimensionfulSpinVectorTag?

Comment (wthrowe) [reply src/ApparentHorizons/Tags.hpp]: They should all have tests, but you don't have to do the others in this PR.

Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp]: It looks like this test passes now, there is one issue, the Radius tag type does not match the function call types. The type is a `DataVector` and the functions call for a `Scalar<DataVector>`. Is that something that needs to be changed and updated for the Radius tag or is it okay to change all the function calls to take a DataVector instead for the radius arguments? @wthrowe

Comment (wthrowe) [reply src/ApparentHorizons/Tags.hpp]: It should probably be a `Scalar<DataVector>`, but let's get @markscheel's input on this.

Comment (markscheel) [reply src/ApparentHorizons/Tags.hpp]: I would be fine if RadiusCompute (and Radius) were changed to `Scalar<DataVector>`.  Other ComputeItems depend on `RadiusCompute` and they should be changed too.  Is the idea to do that change in a subsequent PR?

Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp]: Thank you @markscheel, Should I just implement this change in the current PR? @wthrowe

Comment (wthrowe) [reply src/ApparentHorizons/Tags.hpp]: Let's do it as a separate commit here.  It will be needed to make the test pass.

Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp]: The test is now passing and the RadiusCompute tag was changed to a type `Scalar<DataVector>`. @wthrowe

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (wthrowe) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: What prompted this change?  I don't think anything should have affected the calculation of the Laplacian.

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (wthrowe) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: Remove commented code.

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (wthrowe) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: Avoiding the intermediate DataVector is cleaner.  I.e., `const DataVector used_for_size(ylm_physical_size, std::numeric_limits<double>::signaling_NaN());`  (And I'm pretty sure NaN is the default, but didn't check.)

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (wthrowe) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: "names" -> "named"

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (wthrowe) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: Remove commented code.

**Review by wthrowe — CHANGES_REQUESTED** [review]
Nice.  That's exactly what I had in mind.  Just a few cleanup requests.

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (MarloMo) [reply tests/Unit/ApparentHorizons/Test_Tags.cpp]: That was changed to truncate some of the `spin_vector` results that I was trying to debug in my test and I forgot to undo the change. I'll change that back with the new updates.

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (geoffrey4444) [reply tests/Unit/ApparentHorizons/Test_Tags.cpp]: The components are only initialized to NaN I. Debug mode, so shouldn’t we leave the signaling NaN here?

Comment (wthrowe) [reply tests/Unit/ApparentHorizons/Test_Tags.cpp]: Yes, good point.

**Comment by wthrowe** [comment]
I am happy with this.  Go ahead and squash in the fixup.

**Review by wthrowe — DISMISSED** [review]

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (markscheel) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: [nitpick] why create an extra bool?  It seems simpler and easier to read to just have `CHECK(db::get<StrahlkorperGr...etc);`

**Review by markscheel — DISMISSED** [review]
Looks good!  One small optional suggested change;  Squash immediately if you make that change.

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (MarloMo) [reply tests/Unit/ApparentHorizons/Test_Tags.cpp]: You are right, that is a cleaner edit, change has been made @markscheel

**Review by wthrowe — DISMISSED** [review]

**Review by markscheel — DISMISSED** [review]

File: src/ApparentHorizons/StrahlkorperGr.cpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: Just say "Assert that the DataVectors..."

File: src/ApparentHorizons/StrahlkorperGr.cpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: `get<0>(r_hat)` instead of r_hat.get(0). When you know the component you want to get, spectre provides a compile-time version of get, which will make runtime faster. It only works when you know the component you want to get at compile time (template parameters must be known at compile time, not at runtime)

File: src/ApparentHorizons/StrahlkorperGr.cpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: Similarly, `get<1>(r_hat)` and `get<2>(r_hat)`

File: src/ApparentHorizons/StrahlkorperGr.cpp
Comment (geoffrey4444) [inline src/ApparentHorizons/StrahlkorperGr.cpp]: keep this blank line

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (geoffrey4444) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: delete comment...doesn't really tell you anything that the code itself doesn't say

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]
Just some minor changes...happy for you to squash when others are (I don't need to see this again before approving)

**Review by markscheel — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Comment by wthrowe** [comment]
@geoffrey4444 Can you approve this?  (I can't merge if there is an unresolved change request.)

**Comment by wthrowe** [comment]
@geoffrey4444 You've already approved this in prose, but the merge button is grayed out until you give GitHub a formal check mark.

**Review by geoffrey4444 — APPROVED** [review]

**Comment by geoffrey4444** [comment]
My apologies for the delay! I just approved

**Pull request merged**
